### ISSUE TITLE
Option undefined on Select

### ISSUE THREAD

**Issue by cara62** [Danilo-Js/react-native-ecommerce-components#1]
Hello,

There is an issue with onSelectOption when the search reduce the options. 

This works : 
![Screenshot_20230811_140826_Expo Go](https://github.com/Danilo-Js/react-native-search-select/assets/2953585/36f408bb-6139-47ac-ac41-149f1476fc4c)

after reduce options with search, not works : 

![Screenshot_20230811_140833_Expo Go](https://github.com/Danilo-Js/react-native-search-select/assets/2953585/8e3197ce-33ee-45c0-a382-171541983a83)
![Screenshot_20230811_140842_Expo Go](https://github.com/Danilo-Js/react-native-search-select/assets/2953585/ba6c9cc9-0f7a-4b1c-a052-51cda4328eb9)

with logs, the index is found but not the option 

this is the code : 

```javascript
import * as React from 'react';
import { useState, useEffect } from 'react';
import { View, Alert } from 'react-native';
import SearchSelect from "react-native-search-select";
import Ionicons from 'react-native-vector-icons/Ionicons';
import { ActivityIndicator } from "react-native";

function HomeScreen({navigation}) {
    const [isLoading, setIsLoading] = useState(false);
    const [options, setOptions] = useState([
      {
        key: "0",
        label: "Option 1",
        selected: false,
      },
      {
        key: "1",
        label: "Option 2",
        selected: false,
      },
      {
        key: "2",
        label: "Option 3",
        selected: false,
      },
    ]);

    useEffect((() => {
    }), []);

    return (
      <View style={{ flex: 1, alignItems: 'center', backgroundColor: '#111827'}}>
        {isLoading 
          ? <ActivityIndicator color={"#fff"} style={{justifyContent: 'center', marginTop: '50%'}} /> 
          : <View style={{flex: 1, paddingTop: 35, alignItems: 'center'}}>
          <SearchSelect
              // configuration
              multipleSelect={false}
              options={options}
              setOptions={setOptions}
              onSelectOption={(index, option) => Alert.alert(option.label + ' selected!')}
              // animation
              animationInput="bounceIn"
              animationList="fadeInUpBig"
              // styling
              placeholder="Chercher un mot..."
              searchTextColor="#009E60"
              // icon
              IconSource={Ionicons}
              searchIcon="ios-search-outline"
              searchIconColor="white"
              closeIcon="ios-close-circle"
              closeIconColor="white"
              optionSelectedIcon="ios-checkbox"
              optionSelectedIconColor="#009E60"
              placeholderTextColor="#009E60"
          />
      </View>
        }
      </View>
    );
  }

  export default HomeScreen;
```

Thanks you :) 
Best Regards

**Comment by Danilo-Js** [comment]
Hii !
I have released v3.0.9 fixing this bug.
Thank you so much for finding it.

In case there is still an error, feel free to open another issue.
Have a good day =)

**Comment by cara62** [comment]
@Danilo-Js Thank you so much for your reactivity. 

It's works perfectly now, i was stuck on my project ;) 

Have a good day too !
